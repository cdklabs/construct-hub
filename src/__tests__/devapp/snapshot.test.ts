import '@aws-cdk/assert/jest';

import { App } from '@aws-cdk/core';
import * as cxapi from '@aws-cdk/cx-api';
import { stringify as yaml } from 'yaml';

import { Isolation } from '../../construct-hub';
import { DevStack } from './dev-stack';

expect.addSnapshotSerializer({
  test: (val) => val instanceof cxapi.CloudFormationStackArtifact,
  serialize: (val: cxapi.CloudFormationStackArtifact) => yaml(val.template),
});

test('golden snapshot', () => {
  const app = new App();
  const stack = new DevStack(app, 'dev', { sensitiveTaskIsolation: Isolation.NO_INTERNET_ACCESS });
  expect(app.synth().getStackByName(stack.stackName)).toMatchSnapshot();
});
