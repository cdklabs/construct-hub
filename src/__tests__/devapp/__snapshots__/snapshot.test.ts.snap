// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`golden snapshot 1`] = `
Resources:
  ConstructHubDummyHelloServiceRoleCA7D7B43:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
        Version: 2012-10-17
      ManagedPolicyArns:
        - Fn::Join:
            - ""
            - - "arn:"
              - Ref: AWS::Partition
              - :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
  ConstructHubDummyHello173E8657:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket:
          Ref: AssetParameters82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4bS3Bucket68D54260
        S3Key:
          Fn::Join:
            - ""
            - - Fn::Select:
                  - 0
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParameters82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4bS3VersionKeyDB7FEE81
              - Fn::Select:
                  - 1
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParameters82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4bS3VersionKeyDB7FEE81
      Role:
        Fn::GetAtt:
          - ConstructHubDummyHelloServiceRoleCA7D7B43
          - Arn
      Handler: index.handler
      Runtime: nodejs14.x
    DependsOn:
      - ConstructHubDummyHelloServiceRoleCA7D7B43
  ConstructHubDummyFooServiceRoleD36C3C19:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
        Version: 2012-10-17
      ManagedPolicyArns:
        - Fn::Join:
            - ""
            - - "arn:"
              - Ref: AWS::Partition
              - :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
  ConstructHubDummyFoo7325132B:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket:
          Ref: AssetParametersdaab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5S3BucketEAD14DDE
        S3Key:
          Fn::Join:
            - ""
            - - Fn::Select:
                  - 0
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParametersdaab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5S3VersionKey30AEE575
              - Fn::Select:
                  - 1
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParametersdaab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5S3VersionKey30AEE575
      Role:
        Fn::GetAtt:
          - ConstructHubDummyFooServiceRoleD36C3C19
          - Arn
      Handler: index.handler
      Runtime: nodejs14.x
    DependsOn:
      - ConstructHubDummyFooServiceRoleD36C3C19
  ConstructHubWebAppWebsiteBucket4B2B9DB2:
    Type: AWS::S3::Bucket
    UpdateReplacePolicy: Retain
    DeletionPolicy: Retain
  ConstructHubWebAppWebsiteBucketPolicy17174C06:
    Type: AWS::S3::BucketPolicy
    Properties:
      Bucket:
        Ref: ConstructHubWebAppWebsiteBucket4B2B9DB2
      PolicyDocument:
        Statement:
          - Action: s3:GetObject
            Effect: Allow
            Principal:
              CanonicalUser:
                Fn::GetAtt:
                  - ConstructHubWebAppDistributionOrigin1S3Origin694AF937
                  - S3CanonicalUserId
            Resource:
              Fn::Join:
                - ""
                - - Fn::GetAtt:
                      - ConstructHubWebAppWebsiteBucket4B2B9DB2
                      - Arn
                  - /*
        Version: 2012-10-17
  ConstructHubWebAppDistributionOrigin1S3Origin694AF937:
    Type: AWS::CloudFront::CloudFrontOriginAccessIdentity
    Properties:
      CloudFrontOriginAccessIdentityConfig:
        Comment: Identity for devConstructHubWebAppDistributionOrigin1FBBA04AE
  ConstructHubWebAppDistribution1F181DC9:
    Type: AWS::CloudFront::Distribution
    Properties:
      DistributionConfig:
        CacheBehaviors:
          - CachePolicyId: 658327ea-f89d-4fab-a63d-7e88639e58f6
            Compress: true
            PathPattern: /packages/*
            TargetOriginId: devConstructHubWebAppDistributionOrigin2A726FD66
            ViewerProtocolPolicy: allow-all
        CustomErrorResponses:
          - ErrorCode: 404
            ResponseCode: 200
            ResponsePagePath: /index.html
          - ErrorCode: 403
            ResponseCode: 200
            ResponsePagePath: /index.html
        DefaultCacheBehavior:
          CachePolicyId: 658327ea-f89d-4fab-a63d-7e88639e58f6
          Compress: true
          TargetOriginId: devConstructHubWebAppDistributionOrigin1FBBA04AE
          ViewerProtocolPolicy: allow-all
        DefaultRootObject: index.html
        Enabled: true
        HttpVersion: http2
        IPV6Enabled: true
        Origins:
          - DomainName:
              Fn::GetAtt:
                - ConstructHubWebAppWebsiteBucket4B2B9DB2
                - RegionalDomainName
            Id: devConstructHubWebAppDistributionOrigin1FBBA04AE
            S3OriginConfig:
              OriginAccessIdentity:
                Fn::Join:
                  - ""
                  - - origin-access-identity/cloudfront/
                    - Ref: ConstructHubWebAppDistributionOrigin1S3Origin694AF937
          - CustomOriginConfig:
              OriginProtocolPolicy: https-only
              OriginSSLProtocols:
                - TLSv1.2
            DomainName: awscdk.io
            Id: devConstructHubWebAppDistributionOrigin2A726FD66
  ConstructHubWebAppDeployWebsiteAwsCliLayer23CFFBC1:
    Type: AWS::Lambda::LayerVersion
    Properties:
      Content:
        S3Bucket:
          Ref: AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3BucketAEADE8C7
        S3Key:
          Fn::Join:
            - ""
            - - Fn::Select:
                  - 0
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3VersionKeyE415415F
              - Fn::Select:
                  - 1
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3VersionKeyE415415F
      Description: /opt/awscli/aws
  ConstructHubWebAppDeployWebsiteCustomResourceE6DF98C9:
    Type: Custom::CDKBucketDeployment
    Properties:
      ServiceToken:
        Fn::GetAtt:
          - CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536
          - Arn
      SourceBucketNames:
        - Ref: AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3Bucket11A0D65B
      SourceObjectKeys:
        - Fn::Join:
            - ""
            - - Fn::Select:
                  - 0
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3VersionKey5E1C1FBA
              - Fn::Select:
                  - 1
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3VersionKey5E1C1FBA
      DestinationBucketName:
        Ref: ConstructHubWebAppWebsiteBucket4B2B9DB2
      Prune: true
      DistributionId:
        Ref: ConstructHubWebAppDistribution1F181DC9
    UpdateReplacePolicy: Delete
    DeletionPolicy: Delete
  ConstructHubMonitoringWatchfulDashboardB8493D55:
    Type: AWS::CloudWatch::Dashboard
    Properties:
      DashboardBody:
        Fn::Join:
          - ""
          - - '{"widgets":[{"type":"text","width":24,"height":2,"x":0,"y":0,"properties":{"markdown":"#
              dev/ConstructHub/Dummy/Hello\\n\\n[button:AWS Lambda
              Console](https://console.aws.amazon.com/lambda/home?region='
            - Ref: AWS::Region
            - "#/functions/"
            - Ref: ConstructHubDummyHello173E8657
            - ?tab=graph) [button:CloudWatch
              Logs](https://console.aws.amazon.com/cloudwatch/home?region=
            - Ref: AWS::Region
            - "#logEventViewer:group=/aws/lambda/"
            - Ref: ConstructHubDummyHello173E8657
            - )"}},{"type":"metric","width":6,"height":6,"x":0,"y":2,"properties":{"view":"timeSeries","title":"Invocations/5min","region":"
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Invocations","FunctionName","'
            - Ref: ConstructHubDummyHello173E8657
            - '",{"stat":"Sum"}]],"yAxis":{}}},{"type":"metric","width":6,"height":6,"x":6,"y":2,"properties":{"view":"timeSeries","title":"Errors/5min","region":"'
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Errors","FunctionName","'
            - Ref: ConstructHubDummyHello173E8657
            - '",{"stat":"Sum"}]],"annotations":{"horizontal":[{"label":"Errors
              > 0 for 3 datapoints within 15
              minutes","value":0,"yAxis":"left"}]},"yAxis":{}}},{"type":"metric","width":6,"height":6,"x":12,"y":2,"properties":{"view":"timeSeries","title":"Throttles/5min","region":"'
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Throttles","FunctionName","'
            - Ref: ConstructHubDummyHello173E8657
            - '",{"stat":"Sum"}]],"annotations":{"horizontal":[{"label":"Throttles
              > 0 for 3 datapoints within 15
              minutes","value":0,"yAxis":"left"}]},"yAxis":{}}},{"type":"metric","width":6,"height":6,"x":18,"y":2,"properties":{"view":"timeSeries","title":"Duration/5min","region":"'
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Duration","FunctionName","'
            - Ref: ConstructHubDummyHello173E8657
            - '",{"label":"p99","stat":"p99"}]],"annotations":{"horizontal":[{"label":"p99
              > 2000 for 3 datapoints within 15
              minutes","value":2000,"yAxis":"left"}]},"yAxis":{}}},{"type":"text","width":24,"height":2,"x":0,"y":8,"properties":{"markdown":"#
              dev/ConstructHub/Dummy/Foo\\n\\n[button:AWS Lambda
              Console](https://console.aws.amazon.com/lambda/home?region='
            - Ref: AWS::Region
            - "#/functions/"
            - Ref: ConstructHubDummyFoo7325132B
            - ?tab=graph) [button:CloudWatch
              Logs](https://console.aws.amazon.com/cloudwatch/home?region=
            - Ref: AWS::Region
            - "#logEventViewer:group=/aws/lambda/"
            - Ref: ConstructHubDummyFoo7325132B
            - )"}},{"type":"metric","width":6,"height":6,"x":0,"y":10,"properties":{"view":"timeSeries","title":"Invocations/5min","region":"
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Invocations","FunctionName","'
            - Ref: ConstructHubDummyFoo7325132B
            - '",{"stat":"Sum"}]],"yAxis":{}}},{"type":"metric","width":6,"height":6,"x":6,"y":10,"properties":{"view":"timeSeries","title":"Errors/5min","region":"'
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Errors","FunctionName","'
            - Ref: ConstructHubDummyFoo7325132B
            - '",{"stat":"Sum"}]],"annotations":{"horizontal":[{"label":"Errors
              > 0 for 3 datapoints within 15
              minutes","value":0,"yAxis":"left"}]},"yAxis":{}}},{"type":"metric","width":6,"height":6,"x":12,"y":10,"properties":{"view":"timeSeries","title":"Throttles/5min","region":"'
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Throttles","FunctionName","'
            - Ref: ConstructHubDummyFoo7325132B
            - '",{"stat":"Sum"}]],"annotations":{"horizontal":[{"label":"Throttles
              > 0 for 3 datapoints within 15
              minutes","value":0,"yAxis":"left"}]},"yAxis":{}}},{"type":"metric","width":6,"height":6,"x":18,"y":10,"properties":{"view":"timeSeries","title":"Duration/5min","region":"'
            - Ref: AWS::Region
            - '","metrics":[["AWS/Lambda","Duration","FunctionName","'
            - Ref: ConstructHubDummyFoo7325132B
            - '",{"label":"p99","stat":"p99"}]],"annotations":{"horizontal":[{"label":"p99
              > 2000 for 3 datapoints within 15
              minutes","value":2000,"yAxis":"left"}]},"yAxis":{}}}]}'
      DashboardName: construct-hub
  ConstructHubMonitoringWatchfuldevConstructHubDummyHelloDB98DE41ErrorsAlarmFFE52FED:
    Type: AWS::CloudWatch::Alarm
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 3
      AlarmDescription: Over 0 errors per minute
      Dimensions:
        - Name: FunctionName
          Value:
            Ref: ConstructHubDummyHello173E8657
      MetricName: Errors
      Namespace: AWS/Lambda
      Period: 300
      Statistic: Sum
      Threshold: 0
  ConstructHubMonitoringWatchfuldevConstructHubDummyHelloDB98DE41ThrottlesAlarm3C97A7A0:
    Type: AWS::CloudWatch::Alarm
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 3
      AlarmDescription: Over 0 throttles per minute
      Dimensions:
        - Name: FunctionName
          Value:
            Ref: ConstructHubDummyHello173E8657
      MetricName: Throttles
      Namespace: AWS/Lambda
      Period: 300
      Statistic: Sum
      Threshold: 0
  ConstructHubMonitoringWatchfuldevConstructHubDummyHelloDB98DE41DurationAlarm44566C4D:
    Type: AWS::CloudWatch::Alarm
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 3
      AlarmDescription: p99 latency >= 2s (80%)
      Metrics:
        - Id: m1
          Label: p99
          MetricStat:
            Metric:
              Dimensions:
                - Name: FunctionName
                  Value:
                    Ref: ConstructHubDummyHello173E8657
              MetricName: Duration
              Namespace: AWS/Lambda
            Period: 300
            Stat: p99
          ReturnData: true
      Threshold: 2000
  ConstructHubMonitoringWatchfuldevConstructHubDummyFooFFFE1B3CErrorsAlarmDD38F427:
    Type: AWS::CloudWatch::Alarm
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 3
      AlarmDescription: Over 0 errors per minute
      Dimensions:
        - Name: FunctionName
          Value:
            Ref: ConstructHubDummyFoo7325132B
      MetricName: Errors
      Namespace: AWS/Lambda
      Period: 300
      Statistic: Sum
      Threshold: 0
  ConstructHubMonitoringWatchfuldevConstructHubDummyFooFFFE1B3CThrottlesAlarm93D54889:
    Type: AWS::CloudWatch::Alarm
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 3
      AlarmDescription: Over 0 throttles per minute
      Dimensions:
        - Name: FunctionName
          Value:
            Ref: ConstructHubDummyFoo7325132B
      MetricName: Throttles
      Namespace: AWS/Lambda
      Period: 300
      Statistic: Sum
      Threshold: 0
  ConstructHubMonitoringWatchfuldevConstructHubDummyFooFFFE1B3CDurationAlarm14827D2A:
    Type: AWS::CloudWatch::Alarm
    Properties:
      ComparisonOperator: GreaterThanThreshold
      EvaluationPeriods: 3
      AlarmDescription: p99 latency >= 2s (80%)
      Metrics:
        - Id: m1
          Label: p99
          MetricStat:
            Metric:
              Dimensions:
                - Name: FunctionName
                  Value:
                    Ref: ConstructHubDummyFoo7325132B
              MetricName: Duration
              Namespace: AWS/Lambda
            Period: 300
            Stat: p99
          ReturnData: true
      Threshold: 2000
  CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265:
    Type: AWS::IAM::Role
    Properties:
      AssumeRolePolicyDocument:
        Statement:
          - Action: sts:AssumeRole
            Effect: Allow
            Principal:
              Service: lambda.amazonaws.com
        Version: 2012-10-17
      ManagedPolicyArns:
        - Fn::Join:
            - ""
            - - "arn:"
              - Ref: AWS::Partition
              - :iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
  CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF:
    Type: AWS::IAM::Policy
    Properties:
      PolicyDocument:
        Statement:
          - Action:
              - s3:GetObject*
              - s3:GetBucket*
              - s3:List*
            Effect: Allow
            Resource:
              - Fn::Join:
                  - ""
                  - - "arn:"
                    - Ref: AWS::Partition
                    - ":s3:::"
                    - Ref: AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3Bucket11A0D65B
              - Fn::Join:
                  - ""
                  - - "arn:"
                    - Ref: AWS::Partition
                    - ":s3:::"
                    - Ref: AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3Bucket11A0D65B
                    - /*
          - Action:
              - s3:GetObject*
              - s3:GetBucket*
              - s3:List*
              - s3:DeleteObject*
              - s3:PutObject*
              - s3:Abort*
            Effect: Allow
            Resource:
              - Fn::GetAtt:
                  - ConstructHubWebAppWebsiteBucket4B2B9DB2
                  - Arn
              - Fn::Join:
                  - ""
                  - - Fn::GetAtt:
                        - ConstructHubWebAppWebsiteBucket4B2B9DB2
                        - Arn
                    - /*
          - Action:
              - cloudfront:GetInvalidation
              - cloudfront:CreateInvalidation
            Effect: Allow
            Resource: "*"
        Version: 2012-10-17
      PolicyName: CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF
      Roles:
        - Ref: CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265
  CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756C81C01536:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket:
          Ref: AssetParametersc24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cfS3Bucket55EFA30C
        S3Key:
          Fn::Join:
            - ""
            - - Fn::Select:
                  - 0
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParametersc24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cfS3VersionKey60329B70
              - Fn::Select:
                  - 1
                  - Fn::Split:
                      - "||"
                      - Ref: AssetParametersc24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cfS3VersionKey60329B70
      Role:
        Fn::GetAtt:
          - CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265
          - Arn
      Handler: index.handler
      Layers:
        - Ref: ConstructHubWebAppDeployWebsiteAwsCliLayer23CFFBC1
      Runtime: python3.6
      Timeout: 900
    DependsOn:
      - CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRoleDefaultPolicy88902FDF
      - CustomCDKBucketDeployment8693BB64968944B69AAFB0CC9EB8756CServiceRole89A01265
Outputs:
  ConstructHubWebAppDomainNAmeF5298649:
    Value:
      Fn::GetAtt:
        - ConstructHubWebAppDistribution1F181DC9
        - DomainName
    Export:
      Name: ConstructHubDomainName
  ConstructHubMonitoringWatchfulWatchfulDashboard75D318D0:
    Value:
      Fn::Join:
        - ""
        - - https://console.aws.amazon.com/cloudwatch/home?region=
          - Ref: AWS::Region
          - "#dashboards:name="
          - Ref: ConstructHubMonitoringWatchfulDashboardB8493D55
Parameters:
  AssetParameters82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4bS3Bucket68D54260:
    Type: String
    Description: S3 bucket for asset
      "82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4b"
  AssetParameters82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4bS3VersionKeyDB7FEE81:
    Type: String
    Description: S3 key for asset version
      "82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4b"
  AssetParameters82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4bArtifactHash297EEE8C:
    Type: String
    Description: Artifact hash for asset
      "82d882471cef8cd692b3c0bd36c8deee2c7a6ba4db029780d6b05de0e2c1ad4b"
  AssetParametersdaab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5S3BucketEAD14DDE:
    Type: String
    Description: S3 bucket for asset
      "daab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5"
  AssetParametersdaab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5S3VersionKey30AEE575:
    Type: String
    Description: S3 key for asset version
      "daab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5"
  AssetParametersdaab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5ArtifactHash61547DDB:
    Type: String
    Description: Artifact hash for asset
      "daab642f6006051eef33e6a891c83a6e932bf1a6729c3eacdb7f603e8a7e28e5"
  AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3BucketAEADE8C7:
    Type: String
    Description: S3 bucket for asset
      "e9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68"
  AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68S3VersionKeyE415415F:
    Type: String
    Description: S3 key for asset version
      "e9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68"
  AssetParameterse9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68ArtifactHashD9A515C3:
    Type: String
    Description: Artifact hash for asset
      "e9882ab123687399f934da0d45effe675ecc8ce13b40cb946f3e1d6141fe8d68"
  AssetParametersc24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cfS3Bucket55EFA30C:
    Type: String
    Description: S3 bucket for asset
      "c24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cf"
  AssetParametersc24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cfS3VersionKey60329B70:
    Type: String
    Description: S3 key for asset version
      "c24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cf"
  AssetParametersc24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cfArtifactHash85F58E48:
    Type: String
    Description: Artifact hash for asset
      "c24b999656e4fe6c609c31bae56a1cf4717a405619c3aa6ba1bc686b8c2c86cf"
  AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3Bucket11A0D65B:
    Type: String
    Description: S3 bucket for asset
      "8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410"
  AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410S3VersionKey5E1C1FBA:
    Type: String
    Description: S3 key for asset version
      "8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410"
  AssetParameters8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410ArtifactHash6E9B9D5B:
    Type: String
    Description: Artifact hash for asset
      "8452a12da113d3e02da3aaf4b7c8d6e9cb77bd3a62a09a4105a7a6160cd3f410"

`;
